### PR TITLE
feat: Improve error handling for Install Script

### PR DESCRIPTION
**Pull request by mmomjian**
- Improve error detection for `cd`, `mkdir`
- Add check that `curl` exists and downloads are successful
- Notify when overwriting an existing immich-app folder - may want to prompt for a warning in the future?
- Remove deprecated references to `docker-compose`

### CODE REVIEW

File: docs/docs/install/script.md [hunk 1]
```diff
@@ -17,12 +17,11 @@ curl -o- https://raw.githubusercontent.com/immich-app/immich/main/install.sh | b
 The script will perform the following actions:
 
 1. Download [docker-compose.yml](https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml), and the [.env](https://github.com/immich-app/immich/releases/latest/download/example.env) file from the main branch of the [repository](https://github.com/immich-app/immich).
-2. Populate the `.env` file with necessary information based on the current directory path.
-3. Start the containers.
```
Comment (danieldietzler) [inline docs/docs/install/script.md:21]: I would say that people should still customize their env file. You don't think that's necessary?

**Review by danieldietzler — APPROVED** [review]
LGTM

File: docs/docs/install/script.md (hunk 1 above)
Comment (mmomjian) [reply docs/docs/install/script.md:21]: I wouldn't say that's not necessary, but this script doesn't actually do that. It starts the containers right away so any customizations (such as `DB_PASSWORD` will be too late). We would have to change the behavior of the script to prompt for common common variables and write to the .env if we want to try this.

Comment (danieldietzler) [reply docs/docs/install/script.md:21]: Oh you are correct, fair.

**Pull request merged**